src/MainMenu.cpp

```diff
@@ -1769,4 +1769,256 @@ ZunResult MainMenu::ChoosePracticeLevel()
+                break;
+
+            case CURSOR_OPTIONS_POS_MUSICMODE:
+
+                g_SoundPlayer.PlaySoundByIdx(SOUND_MOVE_MENU, 0);
+                g_Supervisor.StopAudio();
+                if (g_Supervisor.cfg.musicMode <= OFF)
+                {
+                    g_Supervisor.cfg.musicMode = MIDI + 1;
+                }
+                g_Supervisor.cfg.musicMode = g_Supervisor.cfg.musicMode - 1;
+                g_Supervisor.SetupMidiPlayback("bgm/th06_01.mid");
+                g_Supervisor.PlayAudio("bgm/th06_01.mid");
+                break;
+
+            case CURSOR_OPTIONS_POS_PLAYSOUNDS:
+
+                g_SoundPlayer.PlaySoundByIdx(SOUND_MOVE_MENU, 0);
+                if (g_Supervisor.cfg.playSounds <= 0)
+                {
+                    g_Supervisor.cfg.playSounds = 2;
+                }
+                g_Supervisor.cfg.playSounds = g_Supervisor.cfg.playSounds - 1;
+                break;
+
+            case CURSOR_OPTIONS_POS_SCREENMODE:
+
+                g_SoundPlayer.PlaySoundByIdx(SOUND_MOVE_MENU, 0);
+                if (this->windowed <= 0)
+                {
+                    this->windowed = 2;
+                }
+                this->windowed = this->windowed - 1;
+                break;
+            }
+        }
+        if (WAS_PRESSED(TH_BUTTON_MENU | TH_BUTTON_BOMB))
+        {
+            this->cursor = CURSOR_OPTIONS_POS_EXIT;
+            g_SoundPlayer.PlaySoundByIdx(SOUND_BACK, 0);
+        }
+        if (WAS_PRESSED_WEIRD(TH_BUTTON_RIGHT))
+        {
+            switch (this->cursor)
+            {
+            case CURSOR_OPTIONS_POS_LIFECOUNT:
+
+                g_SoundPlayer.PlaySoundByIdx(SOUND_MOVE_MENU, 0);
+                g_Supervisor.cfg.lifeCount = g_Supervisor.cfg.lifeCount + 1;
+                if (g_Supervisor.cfg.lifeCount >= 5)
+                {
+                    g_Supervisor.cfg.lifeCount = 0;
+                }
+                break;
+            case CURSOR_OPTIONS_POS_BOMBCOUNT:
+
+                g_SoundPlayer.PlaySoundByIdx(SOUND_MOVE_MENU, 0);
+                g_Supervisor.cfg.bombCount = g_Supervisor.cfg.bombCount + 1;
+                if (g_Supervisor.cfg.bombCount >= 4)
+                {
+                    g_Supervisor.cfg.bombCount = 0;
+                }
+                break;
+            case CURSOR_OPTIONS_POS_COLORMODE:
+
+                g_SoundPlayer.PlaySoundByIdx(SOUND_MOVE_MENU, 0);
+                g_Supervisor.cfg.colorMode16bit = g_Supervisor.cfg.colorMode16bit + 1;
+                if (g_Supervisor.cfg.colorMode16bit >= 2)
+                {
+                    g_Supervisor.cfg.colorMode16bit = 0;
+                }
+                break;
+            case CURSOR_OPTIONS_POS_MUSICMODE:
+
+                g_SoundPlayer.PlaySoundByIdx(SOUND_MOVE_MENU, 0);
+                g_Supervisor.StopAudio();
+                g_Supervisor.cfg.musicMode = g_Supervisor.cfg.musicMode + WAV;
```

Comment:
```suggestion
                g_Supervisor.cfg.musicMode += 1;
```

It doesn't make much sense to use WAV here IMO. We're just moving by 1 through the enum.

Reply:
You're right, it doesn't, it does make sense with MIDI cause its checking if the cursor its at the end